Header: Simplify the admin bar and style to match global header

ryelle opened the pull request:
Fixes #252. Simplify the admin bar content and style to match the global header. This also enables the admin bar across all sites - or it will, once we remove [mu-plugins/pub/wporg-no-adminbar-if-no-blogs.php](https://github.com/WordPress/wordpress.org/blob/c54eb93f3f27d02bba495f6a418db2316e9a03c6/wordpress.org/public_html/wp-content/mu-plugins/pub/wporg-no-adminbar-if-no-blogs.php).

For logged-in users…

- Remove the W logo & about menu, site name, comments, stats, likes widget, and search.
- Move My Sites dropdown content into the user profile dropdown
- Create a new "Edit" dropdown for "Edit (current object)", "Customize", and "Edit Site" — grouping these to consolidate the menu, but leaving them for fast access for site editors. These only show if relevant to the current site/page/user.
- The "+ New" menu stays as-is, and only shows if you have access to create content
- Shift the "Edit Profile" and "Logout" links to the end of the menu in their own group

For logged-out users…

- Remove all menu items, except Register & Log In
- Re-add Register and Log In if they don't exist

Finally, this styles the admin bar to match the new design library colors (which is why this is bundled as part of `global-header-footer`, so we can easily use those custom properties).

To do in other places:

- [x] Remove [mu-plugins/pub/wporg-no-adminbar-if-no-blogs.php](https://github.com/WordPress/wordpress.org/blob/c54eb93f3f27d02bba495f6a418db2316e9a03c6/wordpress.org/public_html/wp-content/mu-plugins/pub/wporg-no-adminbar-if-no-blogs.php), since the admin bar should be always visible
- [x] Remove [mu-plugins/pub/wporg-admin-bar.php](https://github.com/WordPress/wordpress.org/blob/c54eb93f3f27d02bba495f6a418db2316e9a03c6/wordpress.org/public_html/wp-content/mu-plugins/pub/wporg-admin-bar.php) as those link customizations are no longer needed
- [x] Update `mu-plugins/admin-bar-sandbox.php` to add the sandbox flag as a menu item (?)
- Anything else?

### Screenshots

|  | Mobile | Desktop |
|--------|--------|-------|
| Logged out |  ![450-loggedout](https://user-images.githubusercontent.com/541093/187489604-5b15aa9b-c2ad-4521-a0b0-bda1ee5ef299.png) | ![1320-logged-out](https://user-images.githubusercontent.com/541093/187489862-43454d5d-50cd-471a-9e09-e036d28bcae6.png) |
| Average user | ![450-avguser](https://user-images.githubusercontent.com/541093/187489471-d1880739-7f1c-4ca4-aebe-6583275222e8.png) | ![1320-avguser](https://user-images.githubusercontent.com/541093/187490187-49119f82-8ba6-4719-994f-319a838379dd.png) |
| Average user with Edit | ![450-avguser-w-edit](https://user-images.githubusercontent.com/541093/187490233-fce65e35-ce17-4aec-8463-8b69b06d62fd.png) | ![1020-avguser-w-edit](https://user-images.githubusercontent.com/541093/187490244-b8a177cb-fb6b-4d59-8531-a78231707b49.png) |
| Super admin | ![450-superadmin](https://user-images.githubusercontent.com/541093/187490329-0fe1678c-2e7d-4e48-b99c-fb5fc69be0bb.png) | ![1320-superadmin](https://user-images.githubusercontent.com/541093/187490331-5645b5cc-9483-4f74-b872-876d8b7b1815.png) |

Dropdown examples:

|     | Screenshot |
|---|------------|
| + New menu | <img width="50%" src="https://user-images.githubusercontent.com/541093/187490957-a5ac5cc4-7b8e-4e67-abde-5472da7a9760.png" /> |
| Edit menu | <img width="50%" src="https://user-images.githubusercontent.com/541093/187490950-adeffc35-ecd1-4658-9ea2-673115717283.png" /> |
| Average user | <img width="50%" src="https://user-images.githubusercontent.com/541093/187490949-355854bc-3017-49e5-8e03-0a5de5c895ba.png" /> |
| Network admin user | <img width="50%" src="https://user-images.githubusercontent.com/541093/187490955-c48a4517-97c4-43db-a815-42e6fa135ef5.png" /> |
| Forums | <img width="50%" src="https://user-images.githubusercontent.com/541093/187490951-0442e45f-d415-4868-9c37-a5dc046841bd.png" /> |

This applies to the rosetta sites, too (again with my super-admin user, plus rosetta adds a copy-post plugin).
![1320-superadmin-ja](https://user-images.githubusercontent.com/541093/187489827-5a8746f5-f2e0-4326-896a-e2878f5c52e3.png)

### How to test the changes in this Pull Request:

This ideally needs a sandbox to see the full extent of the changes, but you can test it locally.

If you test on a sandbox, you'll want to delete `/mu-plugins/pub/wporg-no-adminbar-if-no-blogs.php` first.

1. View any page
2. You should see the admin bar, whether logged in or not
3. Log in, you should see a simplified admin bar

I've checked on wordpress.org, rosetta, make.wordpress.org, and subsites; and quick checks on translate.w.org. The admin bar is not used on trac, planet, or codex.

ryelle committed:
Header: Simplify the admin bar and style to match global header

ryelle commented on mu-plugins/blocks/global-header-footer/admin-bar.php:
```diff
@@ -0,0 +1,138 @@
+ * @return bool
+ */
+function should_show_admin_bar( $show_admin_bar ) {
+	// @todo Don't enable on the login page.
```
Currently the admin bar shows up on login.w.org. I'm not sure how to target the login site to exclude it, but also it seems like [this code should be disabling it](https://github.com/WordPress/wordpress.org/blob/c54eb93f3f27d02bba495f6a418db2316e9a03c6/wordpress.org/public_html/wp-content/themes/pub/wporg-login/functions.php#L47). Suggestions welcome 🙂

dd32 replied:
This is because the filter used here runs on the default, and `show_admin_bar()` as used in the login theme only toggles the default state, the filter can still override it.

I wasn't able to duplicate the admin bar showing on login.wordpress.org, but regardless, In https://github.com/WordPress/wordpress.org/commit/99da786734fe0b0c9c82dee870f54f6bbda4c1b4 I've switched the theme to use the filter, which should 100% work to hide it I believe.

ryelle replied:
I don't see it on the login page anymore 👍🏻

ryelle commented on mu-plugins/blocks/global-header-footer/admin-bar.php:
```diff
@@ -0,0 +1,138 @@
+	}
+
+	if ( is_user_logged_in() ) {
+		// Remove all items in these submenus.
+		$parent_remove_list = [ 'wp-logo-external', 'wp-logo', 'site-name', 'appearance' ];
+
+		// Remove all these items (includes the top-level items from the parent list).
+		$remove_list = array_merge(
+			$parent_remove_list,
+			[ 'comments', 'stats', 'search', 'my-sites', 'admin-bar-likes-widget' ]
+		);
```
Specifically removing elements like this ensures that items added by plugins (debug, query monitor, copy post, etc) are not removed.

critterverse commented:
Awesome! Haven't tested but leaving some notes based on the screenshots:

Do we definitely need the arrows + flyout menu for all the "my sites" options? I was imagining that we could remove the flyout menu altogether and just have those links go directly to the site's Dashboard. Also wondering if we need the site icon there — this list would look a lot cleaner without. What do others think?

![arrows](https://user-images.githubusercontent.com/77359364/187534292-47f7ad92-59d8-4178-9016-5fcd6c83ce33.jpg)

Could we also add more spacing in the menus? I think each item could have a cell height of 32px:

<img width="547" alt="menu" src="https://user-images.githubusercontent.com/77359364/187538643-d8f13fec-48c6-4878-93d3-9b9e8d4357ce.png">

Also one small alignment thing, it looks like the display name is inset a little further than the user name:

![alignment](https://user-images.githubusercontent.com/77359364/187538698-3cce0761-6b8e-43e4-98e2-f26dd56cb8c5.jpg)

ryelle commented:
> Do we definitely need the arrows + flyout menu for all the "my sites" options?

I opted not to do that since we were already moving things around, I didn't want to remove _more_ things people might use, but linking off to the dashboard should be enough 👍🏻 

> Also wondering if we need the site icon there — this list would look a lot cleaner without. What do others think?

Agreed, especially since it looks like most sites don't customize it 😆 

> Could we also add more spacing in the menus? I think each item could have a cell height of 32px:

😅  This caused some cascading issues on mobile when I tried it, but I can make another pass.

> Also one small alignment thing, it looks like the display name is inset a little further than the user name:

🤦🏻  totally caused by fixing something else.

ryelle committed:
Fix up spacing and visual bugs

ryelle committed:
Remove the submenu items from sites

ryelle commented:
Okay, I've updated the height of each menu item (and fixed the mobile issue i mentioned), removed the submenus & site icons from the site list, and fixed that name/username alignment issue.

I left the submenu in Network Admin since it's useful to have those shortcuts, and most people won't have that menu.

<img width="316" alt="Screen Shot 2022-08-30 at 5 34 59 PM" src="https://user-images.githubusercontent.com/541093/187547723-94c58bf6-de7d-4650-8196-812bec28b59f.png">

dd32 commented:
I've added this to the list:
> [ ] Remove [mu-plugins/pub/wporg-admin-bar.php](https://github.com/WordPress/wordpress.org/blob/c54eb93f3f27d02bba495f6a418db2316e9a03c6/wordpress.org/public_html/wp-content/mu-plugins/pub/wporg-admin-bar.php)

Testing this change, it seems good to me.

The only issue I can find is, that if you have a long list of blogs, like me, which causes the menu to be longer than the screen height, it's not possible to scroll it / see the bottom actions
_edit: I just realised the CSS from this PR didn't apply properly, if this was already accounted for, just ignore this...._

This is a slightly-hacky suggestion, but one quick-solution to that without needing to have the entire menu scroll (which seems to be hard to do, and actually results in a worse UX anyway IMHO) is this rule:
```
#wp-admin-bar-my-sites-list {
    max-height: 50vh;
    overflow: scroll;
}
```
which results in this: 
https://user-images.githubusercontent.com/767313/187582543-1bf53874-a13f-431c-a543-29348d2cc362.mov

ryelle committed:
Scroll the "My Sites" list if there are many sites

ryelle committed:
Remove todo comment

joedolson commented:
It appears that the 'Edit' options are no longer accessible from the keyboard since this change. @alexstine, the Make/Accessibility team rep noted this because it significantly broke his workflow to not be able to reach the 'Edit Post' submenu while posting agendas & meeting notes. 

I haven't explored thoroughly, but I think it broke the event handlers in /js/_enqueus/lib/admin-bar.js

ryelle commented:
Thanks for the report, can easily reproduce. I'll make a new issue to get that fixed.
